Role permissions modify

unenglishable opened the pull request.

unenglishable committed:
feat(role-permission): stub role permission modify_by_role

akinsey committed:
feat: skeleton for admin role update

unenglishable committed:
docs(role-permission): fix grammar in @doc

unenglishable committed:
docs(seed-roles-permissions): fix spelling in comment

unenglishable committed:
feat(role-permission): implement upsert_modified

used to change the value of modified on a role permission

unenglishable committed:
refactor(role-permission): change input var name permissions -> new_permissions

in modify_by_role
in preparation for updating role's permissions at the end of

unenglishable committed:
feat(role-permission): implement logic for updating role permissions

create new role permissions from old role permissions
upsert new role permissions (modified values only)

unenglishable committed:
feat(role-permission): update permissions for role after modifying

unenglishable committed:
Merge remote-tracking branch 'origin/roles-update' into role-permissions-modify

unenglishable committed:
fix(r-p): remove comma in for

unenglishable committed:
fix(r-p): fix variable names and access value to set new_value

unenglishable committed:
fix(r-p): set modified on old_role_permission via Map.put

unenglishable committed:
fix(r-p): role.id -> role_id

unenglishable committed:
fix(r-p): upsert_modified role_permissions -> roles_permissions

unenglishable committed:
feat(r-p): don't access value from flattened role permissions

value is at the path

unenglishable committed:
fix(r-p): use map instead of modifying RolePermission

update_modified needs a list of maps, not models

unenglishable committed:
fix(r-p): pass permissions to Role.set_permissions

from permissions_map_by_role_id

unenglishable committed:
feat(role-controller): use RolePermission.modify_by_role

unenglishable committed:
feat(r-p): return success

unenglishable committed:
refactor(r-p): set defp on upsert_modified and remove module doc

unenglishable committed:
fix(r-p): permissions_map_by_role_id, set permission true after filter

filtered permissions are true, don't set value from value field on role
permission

unenglishable committed:
fix(r-p): use Enum.reduce instead of for

needed to track the collected result for new_role_permissions

also included, fix collector list append
-> change to prepend (previous way nested list)

akinsey committed:
fix(role-update): authenticate role update route

akinsey committed:
refactor(role-permissions): run mix format

unenglishable committed:
feat(role controller): handle priority restrictions in update

akinsey committed:
feat(validate-helper): implement a way to add default value for validate helper

unenglishable committed:
Merge remote-tracking branch 'origin/main' into role-permissions-modify

unenglishable committed:
Merge remote-tracking branch 'origin/main' into role-permissions-modify

unenglishable committed:
refactor(mix): format

unenglishable committed:
docs(role permission): spelling correction

unenglishable committed:
feat(role controller): implement all/2

unenglishable committed:
feat(router): add /admin/roles/all

unenglishable committed:
test(role controller): add role controller test

unenglishable committed:
test(role controller): refactor, extract roles before assertions

unenglishable committed:
test(role controller): does not get roles when not authenticated

unenglishable committed:
refactor(mix): format

unenglishable committed:
feat(role controller): use id in update

unenglishable committed:
refactor(role controller): temporarily remove priority_restrictions

don't have a handler for this yet

unenglishable committed:
test(role controller): add test for update

check value before and return value of update

unenglishable committed:
test(role controller): correct role id 6 -> 7 (newbie)

unenglishable committed:
test(role controller): check newbie role modification

unenglishable committed:
test(role controller): fix modified newbie permissions for check

use "" => "" instead of key: value

unenglishable committed:
test(role controller): refactor attrs name

unenglishable committed:
refactor(role controller): move params to with

unenglishable committed:
feat(mix): seed admin user for test

unenglishable committed:
docs(session): add todo for input validation

unenglishable committed:
feat(helpers/validate): implement sanitize_list

currently checks if list is list, otherwise returns empty list

useful for when tests pass in nil instead of empty list

unenglishable committed:
refactor(role controller): run priority_restrictions through sanitize_list

unenglishable committed:
test(role_controller): add blank priority restrictions in update test

unenglishable committed:
docs(role model): remove modify_by_role commented pseudocode

unenglishable committed:
feat(role model): implement set_priority_restrictions

updates priority_restrictions list for role

unenglishable committed:
feat(r_p model): call set_priority_restrictions on modify_by_role

unenglishable committed:
feat(role_controller): pass priority_restrictions to modify_by_role

should now update priority_restrictions on api call

unenglishable committed:
style(mix): format

unenglishable committed:
test(role controller): fix authenticated tag placement

should be on tests and not describe

unenglishable committed:
test(role controller): stub priority_restrictions test

unenglishable committed:
feat(role): priority restrictions handler for empty list

unenglishable committed:
test(role controller): test modifying priority_restrictions

unenglishable committed:
test(role controller): fix, pass attrs in body

unenglishable committed:
test(role controller): pass attrs in body

unenglishable committed:
test(role controller): refactor assertion

use == instead of =

unenglishable committed:
test(role controller): reset roles permission back to empty

unenglishable committed:
style(mix): format

unenglishable committed:
refactor(role controller): correct error message

unenglishable committed:
refactor(role permission): underscore unused variable

akinsey commented on lib/epochtalk_server/models/role_permission.ex:
```diff
@@ -59,27 +59,76 @@ defmodule EpochtalkServer.Models.RolePermission do
+  end
+
+  # Used to update the modified field of a `RolePermission` in the database (should already exist)
+  defp upsert_modified([]), do: {:error, "Role permission list is empty"}
```
This probably isn't necessary since this is a private function and you are not checking for the error anywhere

unenglishable replied:
got rid of it ✅

unenglishable committed:
refactor(role permission): remove guard clause for upsert_modified

this code should never be hit, since role permissions is populated
on migration
